### PR TITLE
Adapt utf8 encoding in mailer

### PR DESCRIPTION
**Pull request by busykoala**
Close https://github.com/4teamwork/ogb/issues/6

Mostly because of an issue in outlook 2013 where the encoding header
was displayed in the reply-to field, I adapted the utf-8 encoding here. Also I extracted the
actual sender of the email so that I could better debug.

### CODE REVIEW

File: ftw/subsite/browser/contact.py [hunk 1]
```diff
@@ -121,9 +110,33 @@ def send_feedback(self, recipient, subject, message, sender):
...
-        # send the message
-        mh.send(msg, mto=email_from_email,
-                mfrom=[email_from_email])
+
+            email_from_email = (email_from_email or
+                                reg._records.get('plone.email_from_address').value)
+            email_from_name = (email_from_name or
+                               reg._records.get('plone.email_from_name').value)
+
+        # all the strings fed to send_mail have to be utf-8
```
Comment (mbaechtold) [inline ftw/subsite/browser/contact.py]: What about passing all the arguments of `send_mail()` through `safe_utf8()` in the method itself instead of converting it before passing it to the method? This way the programmer must noch remember to convert the data.

File: ftw/subsite/tests/test_contact.py [hunk 2]
```diff
@@ -33,13 +34,13 @@ def test_contact_everything_ok(self, browser):
...
-        self.assertEqual(msg.get('From'), 'blubber@blubb.ch')
+        self.assertEqual(msg.get('From'), 'Subsite <blubber@blubb.ch>')
         self.assertEqual(msg.get('reply-to'), u'hans peter <test@test.com>')
         sub = msg.get('Subject')
-        self.assertEqual(sub.encode('utf8'), 'Testsubject')
+        self.assertEqual(sub.encode('utf8'), '=?utf-8?q?Testsubject?=')
         self.assertEqual(
             ('hans peter (test@test.com) sends you a message from your site '
-             'Subsite(http:=\n//nohost/plone/subsite):\n'
+             'Subsite (http=\n://nohost/plone/subsite):\n'
```
Comment (mbaechtold) [inline ftw/subsite/tests/test_contact.py]: This is not directly related to your change, but what does the `\n` do in `(http=\n://nohost/plone/subsite)`?

Will it cause a line break between `http` and the remainder of the URL?

Comment (busykoala) [reply ftw/subsite/tests/test_contact.py]: Great Idea! I didn't think about it, narrow minded from a day looking at the same problem.

File: ftw/subsite/browser/contact.py (hunk 1 above)
Comment (busykoala) [reply ftw/subsite/browser/contact.py]: Great Idea! I didn't think about it, narrow minded from a day looking at the same problem.

File: ftw/subsite/tests/test_contact.py (hunk 2 above)
Comment (busykoala) [reply ftw/subsite/tests/test_contact.py]: I'm not certain. It definitely should be a line break because the mail is sent plain text. The whitespace I introduced would also explain why it moved to the front. But why it moved by two chars I don't get as well.

Comment (busykoala) [reply ftw/subsite/tests/test_contact.py]: Relating to our discussion, I think this is file like that. I extracted the change to its own commit to have a better overview.

**Comment by busykoala** [comment]
@mbaechtold, could you please have a second look at it.

File: ftw/subsite/browser/contact.py [hunk 3]
```diff
@@ -121,9 +111,33 @@ def send_feedback(self, recipient, subject, message, sender):
                 # if none is registered on the page in plone5
                 from_email_field._set_value('site@nohost.com')
                 email_from_email = from_email_field.value
-        # send the message
-        mh.send(msg, mto=email_from_email,
-                mfrom=[email_from_email])
+
+            email_from_email = (email_from_email or
+                                reg._records.get('plone.email_from_address').value)
```
[the remark this answers is not in the document]
Comment (busykoala) [reply ftw/subsite/browser/contact.py]: It can be stored at a different location now. Therefore I want to retrieve all information possible before the mailer breaks.

**Comment by busykoala** [comment]
@mbaechtold, I now split the commit even more. The order isn't the same you suggested but I can change that if you'd like. I think like this it is quite logical.

File: development.cfg [hunk 4]
```diff
@@ -1,4 +1,4 @@
 [buildout]
 extends =
-    test-plone-5.1.x.cfg
+    test-plone-4.3.x.cfg
```
Comment (maethu) [inline development.cfg]: Please don't change back to 4.3 :-)

File: ftw/subsite/browser/contact.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (maethu) [reply ftw/subsite/browser/contact.py]: IMGO good addition

File: development.cfg (hunk 4 above)
Comment (busykoala) [reply development.cfg]: Sorry, I changed that for local plone 5 tests and sneaked it in.

**Comment by busykoala** [comment]
@maethu, I added the changelog and additionally removed packages now available for plone 5 from source.
